### DISCUSSION TITLE
More flattering default 3D settings

### DISCUSSION THREAD

**Discussion by AdrianTheFrog** [godotengine/godot-proposals#9602] [3D]
I think a large part of Unreal Engine's reputation as having better graphics than Unity and Godot comes not from better technology but simply making the default "new project" look good. Many users will make and publish quick games or simple demos in the engine without messing with the graphics and because of that people come to associate your game engine's graphics with the default settings rather than what can be accomplished with more tweaking. I think this is where the idea of the "unity look" comes from and a major factor in public opinion especially among less technically-inclined gamers.

I think a large step in the right direction would be to enable antialiasing, some post processing, and SDFGI in the default Godot project as soon as you click "3D scene" with the forward+ renderer. A change in the default skybox would probably also be helpful.

Yes, this could increase lag for users on low end systems creating a new project for the first time. However, I believe these post processing effects are fast enough for this to not be an issue for the vast majority of users.

**Comment by Jesusemora** [comment]
My pc can't handle sdfgi, and I know of people with worse computers.
Godot already has a default sky and sun. What could be done is bundle a sky (4K) from polyhaven or ambientcg (CC 0) and use it as the default sky. This would be enough to make the starting scene look good while not affecting performance much.

**Reply by Calinou** [reply]
> What could be done is bundle a sky (4K) from polyhaven or ambientcg (CC 0) and use it as the default sky.

This would increase editor distribution size by quite a bit, considering you're already looking at bundling a ~10 MB file for something that only some 3D projects will use. This size is generally incompressible as well, as the OpenEXR (or Radiance HDR) format is already compressed.

Remember that the Godot editor is available on Android and web browsers, where a small distribution size is paramount.

There's also the issue that bundling a realistic sky texture would be a bad fit for projects with a pixel art or stylized art direction. In contrast, a plain gradient (as used currently) doesn't have any specific kind of orientation towards a specific artstyle.

**Reply by Jesusemora** [reply]
I realized that later, godot executable is 40mb, the size of some textures, which in hindsight sounds insane.
I find that HDR works faster in godot and has smaller size.

**Reply by AdrianTheFrog** [reply]
Yes, I think physical sky or a tweaked procedural sky is a better option as default.

**Comment by Calinou** [comment]
> Yes, this could increase lag for users on low end systems creating a new project for the first time. However, I believe these post processing effects are fast enough for this to not be an issue for the vast majority of users.

Integrated graphics still struggle a lot with modern rendering effects (even glow which is already enabled by default), so I don't think this is the way to go. Something like https://github.com/godotengine/godot-proposals/issues/2183 could help to an extent, but having this enabled by default could be confusing for users. From their perspective, it might seem like some graphics effects turn themselves off on certain hardware without a clear reason why.

The issue with enabling antialiasing by default is that all techniques have tradeoffs, as explained in the [3D antialiasing](https://docs.godotengine.org/en/latest/tutorials/3d/3d_antialiasing.html) documentation. I don't think we should be making the choice of tradeoffs for the user, as some options like TAA are generally not seen as a good fit for games with a pixel art style (for example).

Remember that in many countries, access to dedicated GPUs is uncommon (due to prices being much higher than in the US or most of Europe).

**Reply by AdrianTheFrog** [reply]
![Screenshot 2024-04-26 165242](https://github.com/godotengine/godot-proposals/assets/69091728/b749ca61-0e8a-408b-9561-818f16662026)

This test scene uses only primitives, and with TAA, tonemap, physical sky, SSR, SSAO, glow, and HDDAGI gets around 440 fps on my 3060 at 1080p. Based on the benchmarks on notebookcheck, that means the hd graphics 510 (basic integrated from 2015) would probably get around 20-40 fps. That's probably enough to hold new users over until they learn how to change the settings.

This really is a tradeoff though, but I think it would probably benefit more users than it hurts. 

Talking about antialiasing specifically, I see that TAA comes with tradeoffs but I think it it works well the majority of the times - again it will probably be good enough as a start, even if users will change it after. This isn't locking them into anything after all.

**Reply by Calinou** [reply]
Your scene has a lot of sky visible, so HDDAGI and SSAO appear less costly than they would be in most projects (with more solid ground covering the screen).

> That's probably enough to hold new users over until they learn how to change the settings.

Remember that when the 3D viewport runs slowly, the entire editor's operation becomes affected. Input lag becomes higher, mouse input becomes less responsive and battery life on laptops becomes shorter. This is why it's important to aim at a 60 FPS baseline even on low-end systems, as typing text or scrolling at 30 FPS feels pretty bad in 2024.

**Reply by MattHorn25** [reply]
> This is why it's important to aim at a 60 FPS baseline even on low-end systems

I definitely agree with this, but I would note that the current default settings are well over 60fps even on most old iGPUs. We don't necessarily need HDDAGI by default on every new project, but including some lower-impact settings could improve things without excessive framerate loss.

**Comment by aXu-AP** [comment]
A good template could be a good alternative to this (maybe an official one or promote some high quality one on project manager) - it wouldn't be the default project so more bells and whistles could be turned on, and as a separate download something like a skybox could be bundled without problem.
Interestingly, I didn't find a single such template in asset library. Most of them add some kind of game managing framework / menu systems or character controller - but none for high quality environment (there's just 2 pages of templates, so it's quick to browse through, altough I might've missed some).
